### ISSUE TITLE
Launching apps on nixos

### ISSUE THREAD

**Issue by ShivanshuKantPrasad** [kmonad/kmonad#679]
I am trying to create keybindings for my web browser, text editor etc.
I set the following in my [config file](https://gist.github.com/ShivanshuKantPrasad/d7a56ef73bf3d8734b958fdc3e577cbc)
```
(defalias web (cmd-button "${pkgs.bash}/bin/bash ${pkgs.brave}/bin/brave"))
(defalias ed (cmd-button "${pkgs.bash}/bin/bash ${pkgs.sublime4}/bin/subl"))
(defalias prn (cmd-button "echo ${pkgs.brave}/bin/brave >> /tmp/brave.txt"))
```
Brave or Sublime doesn't launch but the echo command does work.
`systemctl status kmonad-internal` shows the following when I try to launch brave
```
Feb 07 19:23:43 nixos kmonad-internal-start[31277]: mkdir: cannot create directory ‘/var/empty/.local’: Operation not permitted
Feb 07 19:23:43 nixos kmonad-internal-start[31278]: touch: cannot touch '/var/empty/.local/share/applications/mimeapps.list': No such file or directory
Feb 07 19:23:43 nixos kmonad-internal-start[31280]: chrome_crashpad_handler: --database is required
Feb 07 19:23:43 nixos kmonad-internal-start[31280]: Try 'chrome_crashpad_handler --help' for more information.
Feb 07 19:23:43 nixos kmonad-internal-start[31280]: [31281:31281:0207/192343.711545:ERROR:socket.cc(120)] recvmsg: Connection reset by peer (104)
Feb 07 19:23:43 nixos kmonad-internal-start[31280]: /nix/store/bcijns5f6iq8y8v153dzyascgj9f9fjl-brave-1.47.186/bin/.brave-wrapped: line 48: 31281 Trace/breakpoint trap   (core dumped) "$HER>
```
On the other hand nothing happens when I try to launch sublime.

This problem only happens when kmonad starts with the systemd service created by the nixos module. If I start kmonad on command line with `kmonad config.kbd` then programs launch properly. I assume the problem occurs due to the way nixos isolates packages but I am not sure.

**Comment by KaitlynEthylia** [comment]
I am not familiar with nix so i wont be able help much beyond this point, but if you start kmonad from the commmand line as root do you get the same error? if thats the case you could try prefixing the command with sudo -u `<your user>`,

**Comment by ShivanshuKantPrasad** [comment]
Nope, it works fine from command line even as root. I tried adding `sudo -u username` anyway and it still gives the same error. So, I don't think the problem is related to that.

**Comment by joser93** [comment]
This issue might be a duplicate for this https://github.com/kmonad/kmonad/issues/668. if you restart kmonad systemd service, works?

Actually, as soon I boot my system (Ubuntu), I restart the kmonad.service just to fix this, a workaround not very practical since sometimes I forgot to do that, but still it's a thing.
